crates/ra_hir_ty/src/display.rs

```diff
@@ -9,7 +9,7 @@ pub struct HirFormatter<'a, 'b, DB> {
     fmt: &'a mut fmt::Formatter<'b>,
     buf: String,
     curr_size: usize,
-    max_size: Option<usize>,
+    truncate_options: Option<&'a TruncateOptions>,
```

Comment:
Even if we add `TruncateOptions`, they are `Copy` so it's better to avoid introducing a reference until we *really* need it.

Reply:
> they are Copy

Nope, I did not put a derive on them. 
I'll make them to be one, ok, but I wonder why is it bad to use a reference in such a straighforward case?

Reply:
>I'll make them to be one,

Note if we remove the option completely (and I think we should), we won't need this struct, I think. 

>is it bad to use a reference in such a straighforward case?

It's not really bad, but I, in general, prefer to stick with the simplest thing possible, and even a trivial reference has some complextiy to it.

Reply:
> Note if we remove the option completely (and I think we should), we won't need this struct, I think.

Currently we either need to display the type fully (for hover pop-ups at least) or apply truncations (and be able to enable/disable some of them, now only the size limit can be toggled).

If there will be any other truncations from https://github.com/rust-analyzer/rust-analyzer/issues/1946 that we need to toggle, I think it's better to use the object, since we can put all those toggless there.

But ok, let's use two parameters now then.

Reply:
>(for hover pop-ups at least) 

I'd say that even for hover we want to omit defaults

Reply:
I personally would like to quickly look up the complete type definition somewhere and hovers seemed to be a perfect way of doing this for me.
Any alternatives I can use besides navigating to the actual file and looking it up?

Reply:
I think there's no trivial alternatives, but do you need to lookup a concrete type that often? My hypothesis is that 90% of default parameters is `<S = RandomState>` on HashSets/Map, which seem distracting from the actual bits.

Reply:
>  Any alternatives I can use besides navigating to the actual file and looking it up?

Peek Definition in Code (Ctrl-Shift-F10, I think).

Reply:
> but do you need to lookup a concrete type that often?

Must probably you're right and I don't. Guess I'm a bit uneasy with the fact that the IDE hides that information from me without a choice.

> Peek Definition in Code (Ctrl-Shift-F10, I think).

Alas not:

![image](https://user-images.githubusercontent.com/2690773/71183291-68585880-2280-11ea-93af-246c2ea4240d.png)

But I've done the way you've suggested, no default types in both hints and pop-ups.

Reply:
Ah, fair enough, I was using Peek Definition on the type:

![image](https://user-images.githubusercontent.com/308347/71183508-c9802c00-2280-11ea-863a-69ae346c9a56.png)

Reply:
Oh, right, this way it is actually a nice look up thing.
This way it looks like only something that comes from macros and does not have an explicit type in the source code may cause some issues.

Reply:
Yep, there's Goto Definition and Peek Definition, as well as Goto Type Definition. What we want here is Peek Type Definition, but it is absent at the moment